From: Joel Granados <j.granados@samsung.com>
Subject: memory: remove the now superfluous sentinel element from ctl_table array
Date: Thu, 28 Mar 2024 16:57:48 +0100

This commit comes at the tail end of a greater effort to remove the empty
elements at the end of the ctl_table arrays (sentinels) which will reduce
the overall build time size of the kernel and run time memory bloat by ~64
bytes per sentinel (further information Link :
https://lore.kernel.org/all/ZO5Yx5JFogGi%2FcBo@bombadil.infradead.org/)

Remove sentinel from all files under mm/ that register a sysctl table.

Link: https://lkml.kernel.org/r/20240328-jag-sysctl_remset_misc-v1-1-47c1463b3af2@samsung.com
Signed-off-by: Joel Granados <j.granados@samsung.com>
Reviewed-by: Muchun Song <muchun.song@linux.dev>
Reviewed-by: Miaohe Lin <linmiaohe@huawei.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/compaction.c      |    1 -
 mm/hugetlb.c         |    1 -
 mm/hugetlb_vmemmap.c |    1 -
 mm/memory-failure.c  |    1 -
 mm/oom_kill.c        |    1 -
 mm/page-writeback.c  |    1 -
 mm/page_alloc.c      |    1 -
 7 files changed, 7 deletions(-)

--- a/mm/compaction.c~memory-remove-the-now-superfluous-sentinel-element-from-ctl_table-array
+++ a/mm/compaction.c
@@ -3350,7 +3350,6 @@ static struct ctl_table vm_compaction[]
 		.extra1		= SYSCTL_ZERO,
 		.extra2		= SYSCTL_ONE,
 	},
-	{ }
 };
 
 static int __init kcompactd_init(void)
--- a/mm/hugetlb.c~memory-remove-the-now-superfluous-sentinel-element-from-ctl_table-array
+++ a/mm/hugetlb.c
@@ -5029,7 +5029,6 @@ static struct ctl_table hugetlb_table[]
 		.mode		= 0644,
 		.proc_handler	= hugetlb_overcommit_handler,
 	},
-	{ }
 };
 
 static void hugetlb_sysctl_init(void)
--- a/mm/hugetlb_vmemmap.c~memory-remove-the-now-superfluous-sentinel-element-from-ctl_table-array
+++ a/mm/hugetlb_vmemmap.c
@@ -679,7 +679,6 @@ static struct ctl_table hugetlb_vmemmap_
 		.mode		= 0644,
 		.proc_handler	= proc_dobool,
 	},
-	{ }
 };
 
 static int __init hugetlb_vmemmap_init(void)
--- a/mm/memory-failure.c~memory-remove-the-now-superfluous-sentinel-element-from-ctl_table-array
+++ a/mm/memory-failure.c
@@ -141,7 +141,6 @@ static struct ctl_table memory_failure_t
 		.extra1		= SYSCTL_ZERO,
 		.extra2		= SYSCTL_ONE,
 	},
-	{ }
 };
 
 /*
--- a/mm/oom_kill.c~memory-remove-the-now-superfluous-sentinel-element-from-ctl_table-array
+++ a/mm/oom_kill.c
@@ -724,7 +724,6 @@ static struct ctl_table vm_oom_kill_tabl
 		.mode		= 0644,
 		.proc_handler	= proc_dointvec,
 	},
-	{}
 };
 #endif
 
--- a/mm/page_alloc.c~memory-remove-the-now-superfluous-sentinel-element-from-ctl_table-array
+++ a/mm/page_alloc.c
@@ -6296,7 +6296,6 @@ static struct ctl_table page_alloc_sysct
 		.extra2		= SYSCTL_ONE_HUNDRED,
 	},
 #endif
-	{}
 };
 
 void __init page_alloc_sysctl_init(void)
--- a/mm/page-writeback.c~memory-remove-the-now-superfluous-sentinel-element-from-ctl_table-array
+++ a/mm/page-writeback.c
@@ -2291,7 +2291,6 @@ static struct ctl_table vm_page_writebac
 		.mode		= 0644,
 		.proc_handler	= proc_dointvec_jiffies,
 	},
-	{}
 };
 #endif
 
_
